slice.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+var reAnsi        = require('ansi-regex')
+  , stringifiable = require('es5-ext/object/validate-stringifiable')
+  , length        = require('./get-stripped-length')
+  , sgr           = require('./lib/sgr');
+
+var Token = function Token(token) {
+	this.token = token;
+};
```

Comment:
We actually can't since collection is heterogenous. Ordinary `string`s used for string chunks, `Token`-ish objects is used for tokens (CSI), so I can distinguish them later.
https://github.com/StreetStrider/cli-color/blob/de9255c2ebbe9403ee44ff00da9d16dcfb7723df/slice.js#L48 (different logic for tokens and non-tokens). It's like «newtype».